### PR TITLE
Support system pugixml

### PR DESCRIPTION
**Pull request by Ferenc-**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -314,6 +314,9 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
   set(JANSSON_WITHOUT_TESTS ON)
   add_subdirectory(3party/jansson/jansson/)
   target_include_directories(jansson INTERFACE "${PROJECT_BINARY_DIR}/3party/jansson/jansson/include")
+
+  # Configure pugixml library
```
Comment (biodranik) [inline CMakeLists.txt:318]: What does this comment mean?

**Review by biodranik — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (Ferenc-) [reply CMakeLists.txt:318]: Removed

**Comment by Ferenc-** [comment]
@biodranik Is there anything else contributors can do to move this forward?

File: openlr/openlr_tests/decoded_path_test.cpp [hunk 2]
```diff
@@ -22,7 +22,7 @@
 #include <sstream>
 #include <vector>
 
-#include "3party/pugixml/pugixml/src/pugixml.hpp"
+#include <pugixml.hpp>
 #include "3party/pugixml/utils.hpp"
```
Comment (biodranik) [inline openlr/openlr_tests/decoded_path_test.cpp:26]: Please check and fix this one and other similar includes (if any).

**Review by biodranik — CHANGES_REQUESTED** [review]

File: openlr/openlr_tests/decoded_path_test.cpp (hunk 2 above)
Comment (Ferenc-) [reply openlr/openlr_tests/decoded_path_test.cpp:26]: Could you give specifics, about what you think should to be fixed?
This `utils.hpp` actually does not seem to be part of pugixml, and not even a 3party.
This is the only file which includes it, and it is also unclear why, as apparently none of the functions (neither `XMLToString` nor `DebugPrint` seems to be referenced).
Should it be perhaps moved out of `3party` or if indeed not needed, then perhaps removed?

Comment (biodranik) [reply openlr/openlr_tests/decoded_path_test.cpp:26]: Please remove it if it is not used.

Comment (Ferenc-) [reply openlr/openlr_tests/decoded_path_test.cpp]: Removed.

**Comment by Ferenc-** [comment]
@biodranik For your kind review.

**Review by biodranik — APPROVED** [review]
Thank you!

**Pull request merged**